mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -22,66 +22,196 @@
 import gg.solarmc.loader.Transaction;
 import gg.solarmc.loader.data.DataObject;
 import gg.solarmc.loader.impl.SQLTransaction;
+import gg.solarmc.loader.schema.tables.records.KitpvpKitsOwnershipRecord;
 import gg.solarmc.loader.schema.tables.records.KitpvpStatisticsRecord;
 import org.jooq.DSLContext;
-import org.jooq.impl.DSL;
+import org.jooq.Result;
 
-import java.math.BigDecimal;
-import java.util.Objects;
+import java.util.HashSet;
 import java.util.Set;
 
 import static gg.solarmc.loader.schema.tables.KitpvpStatistics.*;
+import static gg.solarmc.loader.schema.tables.KitpvpKitsOwnership.*;
 
 public class KitPvp implements DataObject {
 
-    private volatile BigDecimal kills;
-    private volatile BigDecimal deaths;
-    private volatile BigDecimal assists;
+    private volatile Integer kills;
+    private volatile Integer deaths;
+    private volatile Integer assists;
 
     private final int userID;
+    private final KitPvpManager manager;
 
-    private Set<Kit> ownedKits;
-
-    public KitPvp(int userID, BigDecimal kills, BigDecimal deaths, BigDecimal assists, Set<Kit> kits) {
+    public KitPvp(int userID, Integer kills, Integer deaths, Integer assists, KitPvpManager manager) {
         this.userID = userID;
 
         this.kills = kills;
         this.deaths = deaths;
         this.assists = assists;
-        this.ownedKits = kits;
+
+        this.manager = manager;
+    }
+
+    private KitpvpStatisticsRecord getStatistics(Transaction transaction) {
+        KitpvpStatisticsRecord record = transaction
+                .getProperty(DSLContext.class)
+                .fetchOne(KITPVP_STATISTICS,KITPVP_STATISTICS.USER_ID.eq(userID));
+
+        assert record != null : "Data is missing!";
+
+        return record;
+    }
+
+    private Result<KitpvpKitsOwnershipRecord> getOwnedKits(Transaction transaction) {
+        //cannot be null, is a list
+
+        return transaction.getProperty(DSLContext.class).fetch(KITPVP_KITS_OWNERSHIP,KITPVP_KITS_OWNERSHIP.USER_ID.eq(userID)); //return all entries where id = this.id
```

Comment:
you don't have to comment what the query does. It should be relatively clear once someone is introduced to jooq

Reply:
This method isn't supposed to exist as there is already a method getKits. I missed this in proofreading.